File: src/SpartanFitness.Application/Common/Results/UserSavesResult.cs
using SpartanFitness.Domain.Aggregates;

namespace SpartanFitness.Application.Common.Results;

public record UserSavesResult(
  List<Muscle> Muscles,
  List<MuscleGroup> MuscleGroups,
  List<Exercise> Exercises,
  List<Workout> Workouts);

File: src/SpartanFitness.ClientApp/src/types/results/UserSavesResult.ts
import Exercise from "../domain/Exercise";
import Muscle from "../domain/Muscle";
import MuscleGroup from "../domain/MuscleGroup";
import Workout from "../domain/Workout";

type UserSavesResult = {
  exercises: Exercise[];
  workouts: Workout[];
  muscles: Muscle[];
  muscleGroups: MuscleGroup[];
};

export default UserSavesResult;
